### ISSUE TITLE
Attempt to update kata-runtime to point to the `main` version

### ISSUE THREAD

**Issue by stevenhorsman** [confidential-containers/cloud-api-adaptor#1596]
As part of the merge to main effort, we have https://github.com/kata-containers/kata-containers/pull/7046 which is adding the remote hypervisor feature to the kata runtime. Once this is merge we should test out whether the CAA can re-vendor on it and see what issues there are. We also know that as part of these changes we want to remove the gogoprotobuf workaround https://github.com/confidential-containers/cloud-api-adaptor/blob/eb1b368f84825bb83b9033f07228e04cdef3ceb1/go.mod#L162-L164 and align with kata where we had to do the same thing.

**Comment by stevenhorsman** [comment]
After a lot of compilation issues, I've got it all compiling not and built a CAA OCI image from it, but when testing it doesn't work:
```
Events:
  Type     Reason                  Age                 From               Message
  ----     ------                  ----                ----               -------
  Normal   Scheduled               35m                 default-scheduler  Successfully assigned default/alpine to sh-libvirt-s390x-e2e-22-04-test-4
  Warning  FailedCreatePodSandBox  7s (x160 over 35m)  kubelet            Failed to create pod sandbox: rpc error: code = Unknown desc = failed to create containerd task: failed to create shim task: remote hypervisor call failed: ttrpc: closed: unknown
```
so I've clearly broken something during the change

**Comment by stevenhorsman** [comment]
Beraldo has recommended re-generating the hypervisor protos with ttrpc rather than grpc, so I've created a branch that has that change in kata-runtime: https://github.com/kata-containers/kata-containers/compare/main...stevenhorsman:hypervisor-ttrpc?expand=1
I'm not reworking the changes to undo some of these related to the ttrpc -> grpc change.

**Comment by stevenhorsman** [comment]
I've updated my branch to use my fork of kata runtime with the ttrpc changes and I think I get further now. as when I try and start a peer pods the error is:
```
Events:
  Type     Reason                  Age                    From               Message
  ----     ------                  ----                   ----               -------
  Normal   Scheduled               7m20s                  default-scheduler  Successfully assigned default/nginx-secret-pod to peer-pods-worker-0
  Warning  FailedCreatePodSandBox  114s (x26 over 7m20s)  kubelet            Failed to create pod sandbox: rpc error: code = Unknown desc = failed to create containerd container: create prepare snapshot dir: failed to create temp dir: stat /var/lib/containerd-nydus/snapshots: no such file or directory: unknown
```

**Comment by stevenhorsman** [comment]
We've been tracking the status on this in a slack thread. A rough summary is:
- After the initial ttrpc switch we still had issues, so re-gen'd the kata hypervisor protos and they seemed to be working better
- This has let us get to the point that the kata-runtime can talk to the CAA as this bit of logs show:
```
2023/11/29 13:48:42 [adaptor/proxy] CreateContainer: containerID:18a7d2fd310f9d92b2ebf495f50bd5f87a3d8a40bd048432493742f456325804
2023/11/29 13:48:42 [adaptor/proxy]     mounts:
2023/11/29 13:48:42 [adaptor/proxy]         destination:/proc source:proc type:proc
2023/11/29 13:48:42 [adaptor/proxy]         destination:/dev source:tmpfs type:tmpfs
2023/11/29 13:48:42 [adaptor/proxy]         destination:/dev/pts source:devpts type:devpts
2023/11/29 13:48:42 [adaptor/proxy]         destination:/dev/mqueue source:mqueue type:mqueue
2023/11/29 13:48:42 [adaptor/proxy]         destination:/sys source:sysfs type:sysfs
2023/11/29 13:48:42 [adaptor/proxy]         destination:/dev/shm source:/run/kata-containers/sandbox/shm type:bind
2023/11/29 13:48:42 [adaptor/proxy]         destination:/etc/resolv.conf source:/run/kata-containers/shared/containers/18a7d2fd310f9d92b2ebf495f50bd5f87a3d8a40bd048432493742f456325804-5c2378f47edc3280-resolv.conf type:bind
2023/11/29 13:48:42 [adaptor/proxy]     annotations:
2023/11/29 13:48:42 [adaptor/proxy]         io.kubernetes.cri.sandbox-cpu-shares: 2
2023/11/29 13:48:42 [adaptor/proxy]         io.katacontainers.pkg.oci.bundle_path: /run/containerd/io.containerd.runtime.v2.task/k8s.io/18a7d2fd310f9d92b2ebf495f50bd5f87a3d8a40bd048432493742f456325804
2023/11/29 13:48:42 [adaptor/proxy]         io.kubernetes.cri.sandbox-cpu-quota: 0
2023/11/29 13:48:42 [adaptor/proxy]         io.kubernetes.cri.sandbox-namespace: default
2023/11/29 13:48:42 [adaptor/proxy]         io.kubernetes.cri.container-type: sandbox
2023/11/29 13:48:42 [adaptor/proxy]         io.kubernetes.cri.sandbox-memory: 0
2023/11/29 13:48:42 [adaptor/proxy]         io.kubernetes.cri.sandbox-name: nginx
2023/11/29 13:48:42 [adaptor/proxy]         io.kubernetes.cri.sandbox-uid: 77aae7ac-3160-4e79-9381-74931196d7b1
2023/11/29 13:48:42 [adaptor/proxy]         nerdctl/network-namespace: /var/run/netns/cni-459f5739-6f42-6c74-6592-c33541e1cfd4
2023/11/29 13:48:42 [adaptor/proxy]         io.kubernetes.cri.sandbox-cpu-period: 100000
2023/11/29 13:48:42 [adaptor/proxy]         io.kubernetes.cri.sandbox-log-directory: /var/log/pods/default_nginx_77aae7ac-3160-4e79-9381-74931196d7b1
2023/11/29 13:48:42 [adaptor/proxy]         io.kubernetes.cri.sandbox-id: 18a7d2fd310f9d92b2ebf495f50bd5f87a3d8a40bd048432493742f456325804
2023/11/29 13:48:42 [adaptor/proxy]         io.katacontainers.pkg.oci.container_type: pod_sandbox
2023/11/29 13:48:42 [adaptor/proxy] Pulling image separately not support on main
2023/11/29 13:48:43 [adaptor/proxy] CreateContainer fails: rpc error: code = Internal desc = failed to mount /run/kata-containers/shared/containers/18a7d2fd310f9d92b2ebf495f50bd5f87a3d8a40bd048432493742f456325804/rootfs to /run/kata-containers/18a7d2fd310f9d92b2ebf495f50bd5f87a3d8a40bd048432493742f456325804/rootfs, with error: ENOENT: No such file or directory
```

It also shows that nydus options aren't being set. This is for at least two reasons:
- kata-deploy in main doesn't support setting the containerd config runtime's snapshotter at the moment. 
- All the nydus mount related changes haven't been merged into main yet.

I think we are now blocked until these steps before we can go much further, but hopefully my kata runtime PR: https://github.com/kata-containers/kata-containers/pull/8520/commits can be merged in the mean time

**Comment by stevenhorsman** [comment]
Just an update on this - now the agent supports image pull on guest I've done a bunch of PoC work on this to push us further in the right direction. The current place we are at is that nydus_snapshotter isn't putting the correct annotation into the storage driver for us to pull on guest. Fabiano is also seeing this on local hypervisor, so hopefully we can work it out between us...

**Comment by stevenhorsman** [comment]
The problem we were hitting in noted in Issue 4 here: https://github.com/kata-containers/kata-containers/issues/8407#issuecomment-2049144827

I ran the following script, kindly provided by Fabiano on the worker:
```
	test_images_to_remove=(
		"docker.io/rancher/mirrored-pause"
		"registry.k8s.io/pause"
		"quay.io/sjenning/nginx"
		"quay.io/prometheus/busybox"
		"quay.io/confidential-containers/test-images"
	)

	ctr_args=""
	if [ "${KUBERNETES}" = "k3s" ]; then
		ctr_args="--address	 /run/k3s/containerd/containerd.sock "
	fi
	ctr_args+="--namespace k8s.io"
	ctr_command="sudo -E ctr ${ctr_args}"
	for related_image in "${test_images_to_remove[@]}"; do
		# We need to delete related image
		image_list=($(${ctr_command} i ls -q |grep "$related_image" |awk '{print $1}'))
		if [ "${#image_list[@]}" -gt 0 ]; then
			for image in "${image_list[@]}"; do
				${ctr_command} i remove "$image"
			done
		fi
		# We need to delete related content of image
		IFS="/" read -ra parts <<< "$related_image"; 
		repository="${parts[0]}";     
		image_name="${parts[1]}";
		formatted_image="${parts[0]}=${parts[-1]}"
		image_contents=($(${ctr_command} content ls | grep "${formatted_image}" | awk '{print $1}'))
		if [ "${#image_contents[@]}" -gt 0 ]; then
			for content in $image_contents; do
				${ctr_command} content rm "$content"
			done
		fi
	done
```
and after that the image pull on host worked and the container is up and running:
```
2024/04/11 09:50:38 [adaptor/proxy]     storages:
2024/04/11 09:50:38 [adaptor/proxy]         mount_point:/run/kata-containers/7ce95eeeb93faef7640d7640c0d46ddd5128b4c81904478fc80ab945fedc4b56/rootfs source:docker.io/library/nginx:latest fstype:overlay driver:image_guest_pull
```
```
# kubectl get pods
NAME    READY   STATUS    RESTARTS   AGE
nginx   1/1     Running   0          76s
```
so we just need a way to do this more easily on the worker node...

**Comment by stevenhorsman** [comment]
After chatting to Pradipta we've realised that we need to change and remove all the CAA install/kustomize's installation of the caa-pod now that the peerpodconfig-ctrl is deploying it. There is a lot of references to it, so I'm trying to go through and unpick and provide alternatives to this...

**Comment by stevenhorsman** [comment]
After a bunch of updates to resolve the double CAA ds the latest instructions are a bit more simplified:

Ok, I'm going to try and describe how to reliably and reproducible set up a dev environment for this for testing. I'm using libvirt with a kcli cluster and an pretty chunky 16 vCPU 32GRAM VM, but I'm not sure that is strictly required to be that large. I will also do some steps like pushing the podvm image that are just so people following that can use my images and save some time:
- Clone my repo (edit this to pick your own if you are based on my branch)
```
export GOPATH="${HOME}/go"
cloud_api_adaptor_repo="github.com/confidential-containers/cloud-api-adaptor"
cloud_api_adaptor_dir="${GOPATH}/src/${cloud_api_adaptor_repo}"
mkdir -p $(dirname "${cloud_api_adaptor_dir}")
git clone -b main "https://${cloud_api_adaptor_repo}.git" "${cloud_api_adaptor_dir}"
pushd $cloud_api_adaptor_dir/src/cloud-api-adaptor
git remote add sh https://github.com/stevenhorsman/cloud-api-adaptor.git
git fetch sh
git checkout -b kata-runtime-bump sh/kata-runtime-bump
```
- Setup libvirt and create a kcli cluster
```
./libvirt/config_libvirt.sh
./libvirt/kcli_cluster.sh create
export KUBECONFIG=$HOME/.kcli/clusters/peer-pods/auth/kubeconfig
```
- Install some pre-reqs
```
sudo snap install yq
yq --version
echo "Install docker"
sudo snap install docker
sudo systemctl start snap.docker.dockerd
sudo systemctl enable snap.docker.dockerd
```
- Build and publish a podvm image (you can skip this and use mine if you want)
```
make podvm-builder podvm-binaries podvm-image
docker image tag quay.io/confidential-containers/podvm-generic-ubuntu-amd64 quay.io/stevenhorsman/podvm-generic-ubuntu-amd64
docker image tag quay.io/confidential-containers/podvm-generic-ubuntu-amd64:c392670e2401a08956ed4f52c4152209617bcde872a71b0a3b87da86b8fad2dc quay.io/stevenhorsman/podvm-generic-ubuntu-amd64:c392670e2401a08956ed4f52c4152209617bcde872a71b0a3b87da86b8fad2dc
docker login quay.io
docker push quay.io/stevenhorsman/podvm-generic-ubuntu-amd64:c392670e2401a08956ed4f52c4152209617bcde872a71b0a3b87da86b8fad2dc
```
- Download the podvm qcow2
```
pushd podvm
./hack/download-image.sh quay.io/stevenhorsman/podvm-generic-ubuntu-amd64:c392670e2401a08956ed4f52c4152209617bcde872a71b0a3b87da86b8fad2dc . -o podvm.qcow2
popd
```
- Prepare the libvirt podvm volume
```
export IMAGE="${PWD}/podvm/podvm.qcow2"
ls -al $IMAGE
virsh -c qemu:///system vol-create-as --pool default --name podvm-base.qcow2 --capacity 20G --allocation 2G --prealloc-metadata --format qcow2
virsh -c qemu:///system vol-upload --vol podvm-base.qcow2 $IMAGE --pool default --sparse
virsh -c qemu:///system vol-info --pool default podvm-base.qcow2
```

- Create a new CAA image (this this optional - you can use mine: `quay.io/stevenhorsman/cloud-api-adaptor:dev-c7c48d677c2c1f4c7f4085c4e663d4a10daf1b2f-dirty`
```
docker login quay.io
registry=quay.io/stevenhorsman make image
```

- Install the operator and libvirt CAA
```
export CAA_IMAGE="quay.io/stevenhorsman/cloud-api-adaptor:dev-c7c48d677c2c1f4c7f4085c4e663d4a10daf1b2f-dirty"
export LIBVIRT_IP="192.168.122.1"
export SSH_KEY_FILE="id_rsa"
./libvirt/install_operator.sh
```
After waiting a little while, you should now have pods:
```
# kubectl get pods -n confidential-containers-system
NAME                                              READY   STATUS              RESTARTS   AGE
cc-operator-controller-manager-7b6c5f84bf-nlqjt   2/2     Running             0          81s
cc-operator-daemon-install-mxxlk                  0/1     ContainerCreating   0          23s
cc-operator-pre-install-daemon-sx4p8              1/1     Running             0          44s
peerpodconfig-ctrl-caa-daemon-j7chc               1/1     Running             0          44s
```
- Log into the worker nodes and clean the pause image cache:
```
kcli ssh peer-pods-worker-0

test_images_to_remove=(
        "docker.io/rancher/mirrored-pause"
        "registry.k8s.io/pause"
        "quay.io/sjenning/nginx"
        "quay.io/prometheus/busybox"
        "quay.io/confidential-containers/test-images"
)

ctr_args=""
if [ "${KUBERNETES}" = "k3s" ]; then
        ctr_args="--address      /run/k3s/containerd/containerd.sock "
fi
ctr_args+="--namespace k8s.io"
ctr_command="sudo -E ctr ${ctr_args}"
for related_image in "${test_images_to_remove[@]}"; do
        # We need to delete related image
        image_list=($(${ctr_command} i ls -q |grep "$related_image" |awk '{print $1}'))
        if [ "${#image_list[@]}" -gt 0 ]; then
                for image in "${image_list[@]}"; do
                        ${ctr_command} i remove "$image"
                done
        fi
        # We need to delete related content of image
        IFS="/" read -ra parts <<< "$related_image";
        repository="${parts[0]}";
        image_name="${parts[1]}";
        formatted_image="${parts[0]}=${parts[-1]}"
        image_contents=($(${ctr_command} content ls | grep "${formatted_image}" | awk '{print $1}'))
        if [ "${#image_contents[@]}" -gt 0 ]; then
                for content in $image_contents; do
                        ${ctr_command} content rm "$content"
                done
        fi
done

exit
```
- Test it by creating a peer pod:
```
echo '
apiVersion: v1
kind: Pod
metadata:
  name: nginx
  annotations:
    io.containerd.cri.runtime-handler: kata-remote
  labels:
    app: nginx
spec:
  runtimeClassName: kata-remote
  containers:
    - name: nginx
      image: nginx
' | kubectl apply -f -
```
After waiting a little while you should see the pod running:
```
# kubectl get pods --watch
NAME                     READY   STATUS              RESTARTS   AGE
nginx-5bb58f7796-lqvr5   0/1     ContainerCreating   0          11s
nginx-5bb58f7796-lqvr5   1/1     Running             0          57s
```
- We can check the CAA logs to see that it was pulled correctly:
```
$ kubectl logs ds/peerpodconfig-ctrl-caa-daemon -n confidential-containers-systemem
+ exec cloud-api-adaptor libvirt -uri 'qemu+ssh://root@192.168.122.1/system?no_verify=1' -data-dir /opt/data-dir -pods-dir /run/peerpod/pods -network-name default -pool-name default -disable-cvm -socket /run/peerpod/hypervisor.sock
cloud-api-adaptor version v0.8.2-dev
  commit: c7c48d677c2c1f4c7f4085c4e663d4a10daf1b2f-dirty
  go: go1.21.9
cloud-api-adaptor: starting Cloud API Adaptor daemon for "libvirt"
2024/04/23 11:05:47 [adaptor/cloud] Cloud provider external plugin loading is disabled, skipping plugin loading
2024/04/23 11:05:47 [adaptor/cloud/libvirt] libvirt config: &libvirt.Config{URI:"qemu+ssh://root@192.168.122.1/system?no_verify=1", PoolName:"default", NetworkName:"default", DataDir:"/opt/data-dir", DisableCVM:true, VolName:"podvm-base.qcow2", LaunchSecurity:"", Firmware:"/usr/share/edk2/ovmf/OVMF_CODE.fd"}
2024/04/23 11:05:47 [adaptor/cloud/libvirt] Created libvirt connection
2024/04/23 11:05:47 [adaptor] server config: &adaptor.ServerConfig{TLSConfig:(*tlsutil.TLSConfig)(0xc0007a0280), SocketPath:"/run/peerpod/hypervisor.sock", CriSocketPath:"", PauseImage:"", PodsDir:"/run/peerpod/pods", ForwarderPort:"15150", ProxyTimeout:300000000000, AAKBCParams:"", EnableCloudConfigVerify:false}
2024/04/23 11:05:47 [util/k8sops] initialized PeerPodService
2024/04/23 11:05:47 [probe/probe] Using port: 8000
2024/04/23 11:05:47 [adaptor] server started
2024/04/23 11:06:01 [podnetwork] routes on netns /var/run/netns/cni-9b568317-31fb-db0a-0a67-c78d789342e0
2024/04/23 11:06:01 [podnetwork]     0.0.0.0/0 via 10.244.1.1 dev eth0
2024/04/23 11:06:01 [podnetwork]     10.244.0.0/16 via 10.244.1.1 dev eth0
2024/04/23 11:06:01 [adaptor/cloud] Credentials file is not in a valid Json format, ignored
2024/04/23 11:06:01 [adaptor/cloud] stored /run/peerpod/pods/f5a5be92bf37cab72c9bba602b047f33ce62a0d9ca4c5f9f4d6e4dc838ae3e0f/daemon.json
2024/04/23 11:06:01 [adaptor/cloud] create a sandbox f5a5be92bf37cab72c9bba602b047f33ce62a0d9ca4c5f9f4d6e4dc838ae3e0f for pod nginx in namespace default (netns: /var/run/netns/cni-9b568317-31fb-db0a-0a67-c78d789342e0)
2024/04/23 11:06:01 [adaptor/cloud/libvirt] LaunchSecurityType: None
2024/04/23 11:06:01 [adaptor/cloud/libvirt] Checking if instance (podvm-nginx-f5a5be92) exists
2024/04/23 11:06:01 [adaptor/cloud/libvirt] Uploaded volume key /var/lib/libvirt/images/podvm-nginx-f5a5be92-root.qcow2
2024/04/23 11:06:01 [adaptor/cloud/libvirt] Create cloudInit iso
2024/04/23 11:06:01 [adaptor/cloud/libvirt] Uploading iso file: podvm-nginx-f5a5be92-cloudinit.iso
2024/04/23 11:06:01 [adaptor/cloud/libvirt] 45056 bytes uploaded
2024/04/23 11:06:01 [adaptor/cloud/libvirt] Volume ID: /var/lib/libvirt/images/podvm-nginx-f5a5be92-cloudinit.iso
2024/04/23 11:06:01 [adaptor/cloud/libvirt] Create XML for 'podvm-nginx-f5a5be92'
2024/04/23 11:06:01 [adaptor/cloud/libvirt] Creating VM 'podvm-nginx-f5a5be92'
2024/04/23 11:06:01 [adaptor/cloud/libvirt] Starting VM 'podvm-nginx-f5a5be92'
2024/04/23 11:06:02 [adaptor/cloud/libvirt] VM id 7
2024/04/23 11:06:23 [adaptor/cloud/libvirt] Instance created successfully
2024/04/23 11:06:23 [adaptor/cloud/libvirt] created an instance podvm-nginx-f5a5be92 for sandbox f5a5be92bf37cab72c9bba602b047f33ce62a0d9ca4c5f9f4d6e4dc838ae3e0f
2024/04/23 11:06:23 [util/k8sops] nginx is now owning a PeerPod object
2024/04/23 11:06:23 [adaptor/cloud] created an instance podvm-nginx-f5a5be92 for sandbox f5a5be92bf37cab72c9bba602b047f33ce62a0d9ca4c5f9f4d6e4dc838ae3e0f
2024/04/23 11:06:23 [tunneler/vxlan] vxlan ppvxlan1 (remote 192.168.122.254:4789, id: 555000) created at /proc/1/task/13/ns/net
2024/04/23 11:06:23 [tunneler/vxlan] vxlan ppvxlan1 created at /proc/1/task/13/ns/net
2024/04/23 11:06:23 [tunneler/vxlan] vxlan ppvxlan1 is moved to /var/run/netns/cni-9b568317-31fb-db0a-0a67-c78d789342e0
2024/04/23 11:06:23 [tunneler/vxlan] Add tc redirect filters between eth0 and vxlan1 on pod network namespace /var/run/netns/cni-9b568317-31fb-db0a-0a67-c78d789342e0
2024/04/23 11:06:23 [adaptor/proxy] Listening on /run/peerpod/pods/f5a5be92bf37cab72c9bba602b047f33ce62a0d9ca4c5f9f4d6e4dc838ae3e0f/agent.ttrpc
2024/04/23 11:06:23 [adaptor/proxy] failed to init cri client, the err: cri runtime endpoint is not specified, it is used to get the image name from image digest
2024/04/23 11:06:23 [adaptor/proxy] Trying to establish agent proxy connection to 192.168.122.254:15150
2024/04/23 11:06:23 [adaptor/proxy] established agent proxy connection to 192.168.122.254:15150
2024/04/23 11:06:23 [adaptor/cloud] agent proxy is ready
2024/04/23 11:06:23 [adaptor/proxy] CreateSandbox: hostname:nginx sandboxId:f5a5be92bf37cab72c9bba602b047f33ce62a0d9ca4c5f9f4d6e4dc838ae3e0f
2024/04/23 11:06:23 [adaptor/proxy]     storages:
2024/04/23 11:06:23 [adaptor/proxy]         mountpoint:/run/kata-containers/sandbox/shm source:shm fstype:tmpfs driver:ephemeral
2024/04/23 11:06:25 [adaptor/proxy] CreateContainer: containerID:f5a5be92bf37cab72c9bba602b047f33ce62a0d9ca4c5f9f4d6e4dc838ae3e0f
2024/04/23 11:06:25 [adaptor/proxy]     mounts:
2024/04/23 11:06:25 [adaptor/proxy]         destination:/proc source:proc type:proc
2024/04/23 11:06:25 [adaptor/proxy]         destination:/dev source:tmpfs type:tmpfs
2024/04/23 11:06:25 [adaptor/proxy]         destination:/dev/pts source:devpts type:devpts
2024/04/23 11:06:25 [adaptor/proxy]         destination:/dev/mqueue source:mqueue type:mqueue
2024/04/23 11:06:25 [adaptor/proxy]         destination:/sys source:sysfs type:sysfs
2024/04/23 11:06:25 [adaptor/proxy]         destination:/dev/shm source:/run/kata-containers/sandbox/shm type:bind
2024/04/23 11:06:25 [adaptor/proxy]         destination:/etc/resolv.conf source:/run/kata-containers/shared/containers/f5a5be92bf37cab72c9bba602b047f33ce62a0d9ca4c5f9f4d6e4dc838ae3e0f-cb97a11b2b31b71e-resolv.conf type:bind
2024/04/23 11:06:25 [adaptor/proxy]     annotations:
2024/04/23 11:06:25 [adaptor/proxy]         io.katacontainers.pkg.oci.container_type: pod_sandbox
2024/04/23 11:06:25 [adaptor/proxy]         io.kubernetes.cri.sandbox-id: f5a5be92bf37cab72c9bba602b047f33ce62a0d9ca4c5f9f4d6e4dc838ae3e0f
2024/04/23 11:06:25 [adaptor/proxy]         io.kubernetes.cri.sandbox-uid: b3ddf27f-a678-4a2b-932e-f286f16c7d03
2024/04/23 11:06:25 [adaptor/proxy]         io.katacontainers.pkg.oci.bundle_path: /run/containerd/io.containerd.runtime.v2.task/k8s.io/f5a5be92bf37cab72c9bba602b047f33ce62a0d9ca4c5f9f4d6e4dc838ae3e0f
2024/04/23 11:06:25 [adaptor/proxy]         io.kubernetes.cri.container-type: sandbox
2024/04/23 11:06:25 [adaptor/proxy]         io.kubernetes.cri.sandbox-cpu-shares: 2
2024/04/23 11:06:25 [adaptor/proxy]         io.kubernetes.cri.sandbox-cpu-quota: 0
2024/04/23 11:06:25 [adaptor/proxy]         io.kubernetes.cri.sandbox-name: nginx
2024/04/23 11:06:25 [adaptor/proxy]         io.kubernetes.cri.sandbox-namespace: default
2024/04/23 11:06:25 [adaptor/proxy]         io.kubernetes.cri.sandbox-log-directory: /var/log/pods/default_nginx_b3ddf27f-a678-4a2b-932e-f286f16c7d03
2024/04/23 11:06:25 [adaptor/proxy]         io.kubernetes.cri.sandbox-memory: 0
2024/04/23 11:06:25 [adaptor/proxy]         io.kubernetes.cri.sandbox-cpu-period: 100000
2024/04/23 11:06:25 [adaptor/proxy]         nerdctl/network-namespace: /var/run/netns/cni-9b568317-31fb-db0a-0a67-c78d789342e0
2024/04/23 11:06:25 [adaptor/proxy]     storages:
2024/04/23 11:06:25 [adaptor/proxy]         mount_point:/run/kata-containers/f5a5be92bf37cab72c9bba602b047f33ce62a0d9ca4c5f9f4d6e4dc838ae3e0f/rootfs source:pause fstype:overlay driver:image_guest_pull
2024/04/23 11:06:25 [adaptor/proxy] StartContainer: containerID:f5a5be92bf37cab72c9bba602b047f33ce62a0d9ca4c5f9f4d6e4dc838ae3e0f
2024/04/23 11:06:27 [adaptor/proxy] CreateContainer: containerID:d666dd01a271f4b3030851e55c39edc037570f3e7f5c148ba5e0b31820eccff8
2024/04/23 11:06:27 [adaptor/proxy]     mounts:
2024/04/23 11:06:27 [adaptor/proxy]         destination:/proc source:proc type:proc
2024/04/23 11:06:27 [adaptor/proxy]         destination:/dev source:tmpfs type:tmpfs
2024/04/23 11:06:27 [adaptor/proxy]         destination:/dev/pts source:devpts type:devpts
2024/04/23 11:06:27 [adaptor/proxy]         destination:/dev/mqueue source:mqueue type:mqueue
2024/04/23 11:06:27 [adaptor/proxy]         destination:/sys source:sysfs type:sysfs
2024/04/23 11:06:27 [adaptor/proxy]         destination:/sys/fs/cgroup source:cgroup type:cgroup
2024/04/23 11:06:27 [adaptor/proxy]         destination:/etc/hosts source:/run/kata-containers/shared/containers/d666dd01a271f4b3030851e55c39edc037570f3e7f5c148ba5e0b31820eccff8-116ec96e0f318291-hosts type:bind
2024/04/23 11:06:27 [adaptor/proxy]         destination:/dev/termination-log source:/run/kata-containers/shared/containers/d666dd01a271f4b3030851e55c39edc037570f3e7f5c148ba5e0b31820eccff8-b1e1de20d924b941-termination-log type:bind
2024/04/23 11:06:27 [adaptor/proxy]         destination:/etc/hostname source:/run/kata-containers/shared/containers/d666dd01a271f4b3030851e55c39edc037570f3e7f5c148ba5e0b31820eccff8-416109fa9381e6d1-hostname type:bind
2024/04/23 11:06:27 [adaptor/proxy]         destination:/etc/resolv.conf source:/run/kata-containers/shared/containers/d666dd01a271f4b3030851e55c39edc037570f3e7f5c148ba5e0b31820eccff8-cec22eb0353bc28d-resolv.conf type:bind
2024/04/23 11:06:27 [adaptor/proxy]         destination:/dev/shm source:/run/kata-containers/sandbox/shm type:bind
2024/04/23 11:06:27 [adaptor/proxy]         destination:/var/run/secrets/kubernetes.io/serviceaccount source:/run/kata-containers/shared/containers/d666dd01a271f4b3030851e55c39edc037570f3e7f5c148ba5e0b31820eccff8-0134a9e1026d5c80-serviceaccount type:bind
2024/04/23 11:06:27 [adaptor/proxy]     annotations:
2024/04/23 11:06:27 [adaptor/proxy]         io.kubernetes.cri.sandbox-id: f5a5be92bf37cab72c9bba602b047f33ce62a0d9ca4c5f9f4d6e4dc838ae3e0f
2024/04/23 11:06:27 [adaptor/proxy]         io.kubernetes.cri.image-name: docker.io/library/nginx:latest
2024/04/23 11:06:27 [adaptor/proxy]         io.kubernetes.cri.container-name: nginx
2024/04/23 11:06:27 [adaptor/proxy]         io.katacontainers.pkg.oci.container_type: pod_container
2024/04/23 11:06:27 [adaptor/proxy]         io.kubernetes.cri.sandbox-name: nginx
2024/04/23 11:06:27 [adaptor/proxy]         io.kubernetes.cri.sandbox-uid: b3ddf27f-a678-4a2b-932e-f286f16c7d03
2024/04/23 11:06:27 [adaptor/proxy]         io.katacontainers.pkg.oci.bundle_path: /run/containerd/io.containerd.runtime.v2.task/k8s.io/d666dd01a271f4b3030851e55c39edc037570f3e7f5c148ba5e0b31820eccff8
2024/04/23 11:06:27 [adaptor/proxy]         io.kubernetes.cri.container-type: container
2024/04/23 11:06:27 [adaptor/proxy]         io.kubernetes.cri.sandbox-namespace: default
2024/04/23 11:06:27 [adaptor/proxy]     storages:
2024/04/23 11:06:27 [adaptor/proxy]         mount_point:/run/kata-containers/d666dd01a271f4b3030851e55c39edc037570f3e7f5c148ba5e0b31820eccff8/rootfs source:docker.io/library/nginx:latest fstype:overlay driver:image_guest_pull
2024/04/23 11:06:31 [adaptor/proxy] StartContainer: containerID:d666dd01a271f4b3030851e55c39edc037570f3e7f5c148ba5e0b31820eccff8
```
- This shows peer pods working using code based just on kata-containers main. You can now clear up the nginx peer pod:
```
kubectl delete pod nginx
```

**Comment by stevenhorsman** [comment]
A small update here - I've created new images with the 3.4 release of kata-containers:
quay.io/stevenhorsman/cloud-api-adaptor:dev-040e9bef8fdb4e2ed94cf68ae04e89076f7b3249
quay.io/stevenhorsman/podvm-generic-ubuntu-amd64:9def86b85afa60d06146608c1f27c74a0568534540f0cc2d60c08cb046cfa0db

**Comment by stevenhorsman** [comment]
This is going okay and I have some e2e passes locally, but can't get them on the e2e CI as the `pull_request_target` only pulls from the `main` branch, so we first have to try and code into the workflow that should work and I've spun up https://github.com/confidential-containers/cloud-api-adaptor/pull/1828 to do this.

I also can't test it on my fork as it runners on the azure runner that I can't access.

**Comment by stevenhorsman** [comment]
The workflow PR https://github.com/confidential-containers/cloud-api-adaptor/pull/1828 has been merged, so hopefully if I rebase the `kata-runtime-bump` branch on that we might be able to get some e2e tests runnings in the PR.
